src/category_theory/FinType.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2020 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import data.fintype.basic
+import data.fin
+import category_theory.concrete_category.bundled
+import category_theory.concrete_category
+import category_theory.full_subcategory
+import category_theory.skeletal
+
+/-!
+# The category of finite types.
+
+We define the category of finite types, denoted `FinType` as
+(bundled) types with a `fintype` instance.
+
+We also define `FinType.Skeleton`, the standard skeleton of `FinType` whose objects are `fin n`
+for `n : ℕ`. We prove that the obvious inclusion functor `FinType.Skeleton ⥤ FinType` is an
+equivalence of categories in `FinType.Skeleton.equivalence`.
+-/
+
+open_locale classical
+open category_theory
+
+/-- The category of finite types. -/
+@[derive has_coe_to_sort]
+def FinType := bundled fintype
```

Comment:
The name `Fintype` might be preferred here, I have no strong preference either way though.

Reply:
Done.